src/che-openshift-authentication.ts

```diff
@@ -11,10 +11,26 @@
 import * as theia from '@theia/plugin';
 import * as che from '@eclipse-che/plugin';
 import { spawn } from 'child_process';
+import axios, { AxiosInstance } from 'axios';
+
+interface Attributes {
+    cluster: string
+}
+
+interface Data {
+    attributes: Attributes
+}
+
+interface OsUserResponse {
+    data: Data[]
+}
 
 export async function start(context: theia.PluginContext) {
     const machineToken = process.env['CHE_MACHINE_TOKEN'];
     const isMultiUser = !!(machineToken && machineToken.length > 0);
+    const axiosInstance: AxiosInstance = axios;
+    const cheApi = process.env['CHE_API'];
+    const isHostedChe = cheApi && cheApi.indexOf('https://che.openshift.io/api') !== -1;
```

Comment:
Can we somehow avoid hardcoding of domain name here?

Reply:
We can make an API request but I think it is not a good solution.